rest-service/manager_rest/test/security/test_userstore_reload.py

```diff
@@ -0,0 +1,68 @@
+#########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+import tempfile
+import yaml
+import time
+
+from cloudify_rest_client.exceptions import UserUnauthorizedError
+from nose.plugins.attrib import attr
+
+from manager_rest.test import base_test
+from manager_rest.test.security.security_test_base import SecurityTestBase
+
+_, USERSTORE_FILE = tempfile.mkstemp()
+
+
+@attr(client_min_version=2, client_max_version=base_test.LATEST_API_VERSION)
+class TestUserstoreReloadFile(SecurityTestBase):
+
+    def setUp(self):
+        self.populate_userstore_file(self.get_users(), self.get_groups())
+        super(TestUserstoreReloadFile, self).setUp()
+
+    def populate_userstore_file(self, users, groups):
+        userstore_settings = {
+            'users': users,
+            'groups': groups
+        }
+        with open(USERSTORE_FILE, 'w') as outfile:
+            outfile.write(yaml.dump(userstore_settings,
+                                    default_flow_style=True))
+
+    def test_reload_userstore(self):
+        client = self.create_client(headers=SecurityTestBase.
+                                    create_auth_header(
+                                        username='alice',
+                                        password='alice_password'))
+        # alice should be able to do everything
+        client.deployments.list()
+
+        # modify the userstore file
+        self.populate_userstore_file([], [])
```

Comment:
maybe update with a different role for alice? when you empty the entire userstore we might not be testing exactly that...